src/trading/TakeOrder.sol

```diff
@@ -0,0 +1,393 @@
+        uint256 _sharesToSell = getTakerSharesToSell(_contracts.longShareToken, _contracts.shortShareTokens, _address, _direction, _size);
+        uint256 _sharesToBuy = _size.sub(_sharesToSell);
+        return Participant({
+            participantAddress: _address,
+            direction: _direction,
+            startingSharesToSell: _sharesToSell,
+            startingSharesToBuy: _sharesToBuy,
+            sharesToSell: _sharesToSell,
+            sharesToBuy: _sharesToBuy
+        });
+    }
+
+    function getTokensToCover(Direction _direction, uint256 _sharePriceRange, uint256 _sharePriceLong, uint256 _sharePriceShort, uint256 _numShares) internal constant returns (uint256) {
+        return _numShares
+            .mul((_direction == Direction.Long) ? _sharePriceLong : _sharePriceShort)
+            .div(_sharePriceRange)
+        ; // move semicolon up a line when https://github.com/duaraghav8/Solium/issues/110 is fixed
+    }
+
+    function getShortShareTokens(IMarket _market, uint8 _longOutcome) private constant returns (IShareToken[] memory) {
+        IShareToken[] memory _shortShareTokens = new IShareToken[](_market.getNumberOfOutcomes() - 1);
+        for (uint8 _outcome = 0; _outcome < _shortShareTokens.length + 1; ++_outcome) {
+            if (_outcome == _longOutcome) {
+                continue;
+            }
+            uint8 _index = (_outcome < _longOutcome) ? _outcome : _outcome - 1;
+            _shortShareTokens[_index] = _market.getShareToken(_outcome);
+        }
+        return _shortShareTokens;
+    }
+
+    function getSharePriceDetails(IMarket _market, IOrders _orders, bytes32 _orderId) private constant returns (uint256 _sharePriceRange, uint256 _sharePriceLong, uint256 _sharePriceShort) {
+        int256 _maxDisplayPrice = _market.getMaxDisplayPrice();
+        int256 _minDisplayPrice = _market.getMinDisplayPrice();
+        int256 _orderDisplayPrice = _orders.getPrice(_orderId);
+        _sharePriceRange = uint256(_maxDisplayPrice - _minDisplayPrice);
+        _sharePriceLong = uint256(_orderDisplayPrice - _minDisplayPrice);
+        _sharePriceShort = uint256(_maxDisplayPrice - _orderDisplayPrice);
+        return (_sharePriceRange, _sharePriceLong, _sharePriceShort);
+    }
+
+    function getDirections(IOrders _orders, bytes32 _orderId) private constant returns (Direction _makerDirection, Direction _takerDirection) {
+        return (_orders.getTradeType(_orderId) == Order.TradeTypes.Bid)
+            ? (Direction.Long, Direction.Short)
+            : (Direction.Short, Direction.Long)
+        ; // move semicolon up a line when https://github.com/duaraghav8/Solium/issues/110 is fixed
+    }
+
+    function getTakerSharesToSell(IShareToken _longShareToken, IShareToken[] memory _shortShareTokens, address _taker, Direction _takerDirection, uint256 _takerSize) private constant returns (uint256) {
+        uint256 _sharesAvailable = 2**256-1;
```

Comment:
Perhaps instead of hard-coding this value, it should be added as a constant to SafeMathUint256.sol?

Reply:
Fixed.